consumptions/tests/test_views.py

```diff
@@ -1,32 +1,43 @@
-from django.test import TestCase, Client
+from django.test import TestCase
 from django.urls import reverse
-from consumptions.models import *
 import json
+from rest_framework import status
+from rest_framework.test import APIClient
+from consumptions.models import Product_Category, Product
+
 
 class TestViews(TestCase):
 
     def setUp(self):
-        self.client = Client()
-        self.prod_cat = Product_Category.objects.create(code='LL', name='landline')
-        self.prod = Product.objects.create(code='LL001', name='ADSL', category=self.prod_cat)
-        self.cons = Consumption.objects.create(product=self.prod, quantity=10)
+        self.client = APIClient()
 
     def test_index_get(self):
-        response = self.client.get(reverse('consumptions:index'))
-        self.assertEquals(response.status_code, 200)
-        self.assertTemplateUsed(response, 'index.html')
+        """Test retrieving the home page"""
+        res = self.client.get(reverse('consumptions:index'))
+        self.assertEqual(res.status_code, status.HTTP_200_OK)
+        self.assertTemplateUsed(res, 'index.html')
 
     def test_consumption_table_get(self):
-        response = self.client.get(reverse('consumptions:consumption_list'))
-        self.assertEquals(response.status_code, 200)
-        self.assertTemplateUsed(response, 'consumption_list.html')
+        """Test retrieving a consumptions list"""
+        res = self.client.get(reverse('consumptions:consumption_list'))
+        self.assertEqual(res.status_code, status.HTTP_200_OK)
+        self.assertTemplateUsed(res, 'consumption_list.html')
 
     def test_consumption_create_get(self):
-        response = self.client.get(reverse('consumptions:consumption_create'))
-        self.assertEquals(response.status_code, 200)
-        self.assertTemplateUsed(response, 'consumption_form.html')
-
+        """Test retrieving the create view"""
+        res = self.client.get(reverse('consumptions:consumption_create'))
```

Comment:
how come we are calling `client.get`? since this is a `create` action I would expect as per REST conventions that this endpoint responds only to POST.

Reply:
this test refers to the create view not to the create action 

[](url)

Reply:
![create-view](https://user-images.githubusercontent.com/38106051/106900636-07dfe100-66f7-11eb-8216-37ba6d9c4984.png)